### PR TITLE
Improve help threads

### PR DESCRIPTION
**Pull request by Soheab**
This PR does the following:

- Disallow timed-out users to open/close threads.
- Change the nextcord button style to blurple instead of red (why was it red?)
- Indefinitely fixes the close button (finally)
- Fixes an error that could appear when closing threads (awaiting .history)
- And now to the main change...

Implemented a wait_for that waits 30 minutes for the thread author to send a message else close it before pinging all the helpers.

- An ext.tasks.Loop that checks every 24 hours for empty threads and mentions the helper roles if needed or closes it.

### CODE REVIEW

File: cogs/help.py [hunk 1]
```diff
@@ -89,23 +92,39 @@ async def create_help_thread(self, interaction: Interaction) -> None:
...
         type_to_colour: Dict[str, Colour] = {
-            "Nextcord": Colour.red(),
+            "Nextcord": Colour.blurple(),
             "Python": Colour.green()
         }
 
         em = Embed(
             title = f"{self._help_type} Help needed!",
-            description = f"Alright now that we are all here to help, what do you need help with?",
+            description = f"What do you need help with?",
```
Comment (parker02311) [inline cogs/help.py:101]: Was this change really necessary?

File: cogs/help.py [hunk 2]
```diff
@@ -89,23 +92,39 @@ async def create_help_thread(self, interaction: Interaction) -> None:
...
-            content = f"<@&{HELPER_ROLE_ID}> | {interaction.user.mention}",
+            content = interaction.user.mention,
             embed = em,
             view = ThreadCloseView()
         )
         await msg.pin(reason = "First message in help thread with the close button.")
+        return thread
+
+    async def __launch_wait_for_message(self, thread: Thread, interaction: Interaction) -> None:
+        def yeet(message: Message) -> bool:
```
Comment (parker02311) [inline cogs/help.py:115]: This should be named more professionally

Maybe `is_allowed`

File: cogs/help.py [hunk 3]
```diff
@@ -114,8 +133,8 @@ def disable_all_buttons():
             for _item in confirm_view.children:
                 _item.disabled = True
 
-        confirm_content = "Are you really sure you want to make a help thread?"
-        await interaction.response.send_message(content = confirm_content, ephemeral = True, view = confirm_view)
+        confirm_content = f"Are you really sure you want to make a {self._help_type} Help thread?"
```
Comment (parker02311) [inline cogs/help.py:136]: `Help` should have a lowercase H

File: cogs/help.py [hunk 4]
```diff
@@ -124,15 +143,22 @@ def disable_all_buttons():
...
+    def __init__(self, bot: commands.Bot):
         super().__init__(timeout = None)
-        self.add_item(HelpButton("Nextcord", style = ButtonStyle.red, custom_id = "nextcord"))
-        self.add_item(HelpButton("Python", style = ButtonStyle.green, custom_id = "python"))
+        self.add_item(HelpButton(bot, "Nextcord", style = ButtonStyle.blurple, custom_id = "nextcord"))
+        self.add_item(HelpButton(bot, "Python", style = ButtonStyle.green, custom_id = "python"))
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        if interaction.user.timeout is not None:
+            await interaction.send("You are currently timed out. Please wait until you are removed from it.", ephemeral=True)
```
Comment (parker02311) [inline cogs/help.py:158]: This should be worded better.

For example: `You are currently timed out. Please wait until your time out expires before trying again.`

File: cogs/help.py [hunk 5]
```diff
@@ -181,7 +207,15 @@ async def interaction_check(self, interaction: Interaction) -> bool:
         if not isinstance(interaction.channel, Thread) or interaction.channel.parent_id != HELP_CHANNEL_ID:
             return False
 
-        return interaction.user.id == self._thread_author.id or interaction.user.get_role(HELP_MOD_ID)
+        if interaction.user.timeout is not None:
+            await interaction.send("You are currently timed out. Please wait until you are removed from it.", ephemeral=True)
```
Comment (parker02311) [inline cogs/help.py:211]: ^

**Review by parker02311 — CHANGES_REQUESTED** [review]
Grammar Suggestions

File: cogs/help.py (hunk 2 above)
Comment (Soheab) [reply cogs/help.py:115]: ty, i couldn't think of anything other than `check`.

File: cogs/help.py (hunk 1 above)
Comment (parker02311) [reply cogs/help.py:101]: Yes it was

**Review by parker02311 — APPROVED** [review]
LGTM

**Review by VincentRPS — APPROVED** [review]

**Review by TAG-Epic — APPROVED** [review]

**Review by koala9712 — CHANGES_REQUESTED** [review]
[Method is taken but not accessed?](https://github.com/nextcord/previous/pull/49/files#diff-e800b0746410bdf0a85d6c9986fb44917d89bac6157a5c02dc8bd162c24a2fb4R44)



[Include who closed the thread? and how/why (perhaps use method)](https://github.com/nextcord/previous/pull/49/files#diff-e800b0746410bdf0a85d6c9986fb44917d89bac6157a5c02dc8bd162c24a2fb4R68)

File: cogs/help.py [hunk 6]
```diff
@@ -42,14 +49,19 @@ async def close_help_thread(method: str, thread_channel, thread_author):
     """Closes a help thread. Is called from either the close button or the
     =close command.
     """
+
+    # no need to do any of this if the thread is already closed.
+    if (thread_channel.locked or thread_channel.archived):
+        return
+
+    print(method)
```
Comment (TAG-Epic) [inline cogs/help.py:57]: Debug print left there by accident?

File: cogs/help.py [hunk 7]
```diff
@@ -89,23 +102,39 @@ async def create_help_thread(self, interaction: Interaction) -> None:
...
             "Python": Colour.green()
         }
 
         em = Embed(
             title = f"{self._help_type} Help needed!",
-            description = f"Alright now that we are all here to help, what do you need help with?",
+            description = f"What do you need help with?",
             colour = type_to_colour.get(self._help_type, Colour.blurple())
         )
-        em.set_footer(text = "You and the helpers can close this thread with the button")
```
Comment (TAG-Epic) [inline cogs/help.py:101]: They are referred to as helpers again.

**Review by TAG-Epic — CHANGES_REQUESTED** [review]
Also does not seem to handle reboots.

**Comment by Soheab** [comment]
> Also does not seem to handle reboots.

[it does?](https://github.com/nextcord/previous/pull/49/files#diff-e800b0746410bdf0a85d6c9986fb44917d89bac6157a5c02dc8bd162c24a2fb4R266-R292)
